pkg/build/api/types.go

```diff
@@ -559,10 +566,35 @@ type BuildConfigSpec struct {
 	// are defined, a new build can only occur as a result of an explicit client build creation.
 	Triggers []BuildTriggerPolicy
 
+	// SchedulePolicy describes how the new build created from this build
+	// configuration will be scheduled for execution.
+	SchedulePolicy BuildSchedulePolicy
+
 	// BuildSpec is the desired build specification
 	BuildSpec
 }
 
+// BuildSchedulePolicy defines the behaviour of how the new builds are executed
+// from the existing build configuration.
+type BuildSchedulePolicy string
+
+const (
+	// BuildSchedulePolicyParallel schedules new builds immediately after they are
+	// created. Builds will be executed in parallel. This is the default policy.
+	BuildSchedulePolicyParallel BuildSchedulePolicy = "parallel"
+
+	// BuildSchedulePolicySerial schedules new builds to execute in a sequence as
+	// they are created. Every build gets queued up and will execute when the
+	// previous build completes.
+	BuildSchedulePolicySerial BuildSchedulePolicy = "serial"
+
+	// BuildSchedulePolicyRecent schedules new builds to execute in a sequence as
+	// they are created. Every build gets queued up and will execute when the
+	// previous build completes. Users can mark a build for an immediate execution
+	// using the SkipBuildsInQueue option on the build request.
+	BuildSchedulePolicyRecent BuildSchedulePolicy = "recent"
```

Comment:
@smarterclayton @bparees not sure if "recent" is the right description of the behavior here. Also not 100% sure we even need this policy. Can't it just be the "serial" option? IOW. if I have "serial" policy, I can always clean up the queue by creating a build request that wipes out the queue.